File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cs
public class Solution {
    public int FindDuplicate(int[] a) {
        int n = a.Length;
        int [] freq = new int [n + 2];
        for (int i = 0;i < n + 2; i++){
            freq[i] = 0;
        }
        
        
        foreach (var i in a){
            if (freq[i] == 1) return i;
            freq[i] ++;
        }
        return 0;
    }
}

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:

    int findDuplicate(vector<int>& a) {
        int n = a.size();
        vector <int> freq(n + 2 , 0);
        
        for (auto i : a){
            if (freq[i] >= 1) return i;
            freq[i] ++;
        }
        return 0;
    }
};